bagitobjecttransfer/recordtransfer/static/recordtransfer/js/transferform/forms.js

```diff
@@ -530,6 +567,71 @@ $(() => {
     if (typeof MODAL_MODE !== 'undefined' && MODAL_MODE) {
         initializeModalMode();
     }
+
+
+    /**********************************************************************************************
+     * Source information form setup
+     *********************************************************************************************/
+
+    // Context passed from template to JS
+    const sourceInfoContextElement = document.getElementById("py_context_source_info");
+
+    if (sourceInfoContextElement) {
+        const sourceInfoContext = JSON.parse(sourceInfoContextElement.textContent);
+
+        // Toggles Other source type field depending on value of Source type field
+        setupSelectOtherToggle2(
+            sourceInfoContext["id_source_type"],
+            sourceInfoContext["id_other_source_type"],
+            sourceInfoContext["other_type_id"]
+        );
+
+        // Toggles Other source role field depending on value of Source role field
+        setupSelectOtherToggle2(
+            sourceInfoContext["id_source_role"],
+            sourceInfoContext["id_other_source_role"],
+            sourceInfoContext["other_role_id"]
+        );
+
+        // Add faux-required-field to labels
+        document.querySelectorAll(".faux-required-field").forEach((el) => {
+            let label = el.previousElementSibling;
+            if (label && label.tagName === "LABEL") {
+                label.classList.add("faux-required-field");
+            }
+        });
+
+        const sourceTypeSelect = document.getElementById(sourceInfoContext["id_source_type"]);
+        const sourceRoleSelect = document.getElementById(sourceInfoContext["id_source_role"]);
+        const enterManualInfoInputId = sourceInfoContext["id_enter_manual_source_info"];
+        const enterManualInfoInputElement = document.getElementById(enterManualInfoInputId);
+
+        if (enterManualInfoInputElement) {
+            enterManualInfoInputElement.addEventListener("change", function (event) {
+                const selected = this.querySelector("input[type=radio]:checked").value;
+
+                if (selected === "yes") {
+                    document.querySelectorAll(".initially-hidden").forEach((el) => {
+                        el.classList.remove("hidden-item");
+                    });
+
+                    // Dispatch events to update "other" fields if they were just shown by accident
+                    sourceTypeSelect.dispatchEvent(new Event("change"));
```

Comment:
I also get why you need to trigger an event to update the hidden status of the other source role and other source type fields. I was initially confused by why there were two separate event listeners - one for when the user changes whether they are entering the source info manually (Yes/No) and one for whenever the select value of the dropdown for other source/type changes

Reply:
I wonder which is better practice:

1) (current) Defining a function inline to call upon event trigger and then triggering event
```javascript
enterManualInfoInputElement.addEventListener("change", function (event) {
// contents
}

enterManualInfoInputElement.dispatchEvent(new Event("change")); # manually triggering event 
```
2) Defining function outside, and referencing it as the the function to call when defining an event listener:
```javascript
def updateHiddenFormFields() {}
enterManualInfoInputElement.addEventListener("change", updateHiddenFormFields)

updateHiddenFormField() // Manually calling function for the first time upon load
```

Reply:
Unfortunately I'm using `this` inside the function, which gives me a direct reference to the element where the event originated from.

https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/blob/f0e560c8143537e8a24a07e15051c73d125f9f3d/bagitobjecttransfer/recordtransfer/static/recordtransfer/js/transferform/forms.js#L611

If I call the function outside of an event listener I lose the ability to use `this`.

Reply:
I see, this makes sense in that case.